src/main/java/hexgui/gui/HexGui.java

```diff
@@ -0,0 +1,2393 @@
+  }
+
+  private void showDfpnBounds(String str) {
+    Vector<Pair<String, String>> pairs = StringUtils.parseStringPairList(str);
+    for (int i = 0; i < pairs.size(); i++) {
+      HexPoint point = HexPoint.get(pairs.get(i).first);
+      String value = pairs.get(i).second;
+      m_guiboard.setText(point, value);
+      if (value.trim().equals("W")) m_guiboard.setAlphaColor(point, Color.green);
+      else if (value.trim().equals("L")) m_guiboard.setAlphaColor(point, Color.red);
+    }
+  }
+
+  /** Draws the inferior cells to the gui board. */
+  private void showInferiorCells(String str) {
+    Vector<Pair<String, String>> pairs = StringUtils.parseStringPairList(str);
+    for (int i = 0; i < pairs.size(); i++) {
+      HexPoint point = HexPoint.get(pairs.get(i).first);
+      String value = pairs.get(i).second;
+
+      if (value.charAt(0) == 'f') // fill-in
+      {
+        assert (3 == value.length());
+        if (value.charAt(1) == 'd') // dead
+        m_guiboard.setAlphaColor(point, Color.cyan);
+        else if (value.charAt(1) == 'p') // permanently inferior
+        m_guiboard.setAlphaColor(point, Color.gray);
+        else // captured
+        {
+          assert (value.charAt(1) == 'c');
+          m_guiboard.setAlphaColor(point, Color.red);
+        }
+        if (value.charAt(2) == 'b') m_guiboard.setColor(point, HexColor.BLACK);
+        else {
+          assert (value.charAt(2) == 'w');
+          m_guiboard.setColor(point, HexColor.WHITE);
+        }
+      } else if (value.charAt(0) == 'i') // ignorable
+      {
+        assert (4 <= value.length());
+        if (value.charAt(1) == 'v') // vulnerable
+        m_guiboard.setAlphaColor(point, Color.green);
+        else if (value.charAt(1) == 'r') // reversible
+        m_guiboard.setAlphaColor(point, Color.magenta);
+        else // dominated
+        {
+          assert (value.charAt(1) == 'd');
+          m_guiboard.setAlphaColor(point, Color.yellow);
+        }
+        assert (value.charAt(2) == '[' && value.charAt(value.length() - 1) == ']');
+        String pts = value.substring(3, value.length() - 1);
+        Vector<HexPoint> pp = StringUtils.parsePointList(pts, "-");
+        for (int j = 0; j < pp.size(); ++j) m_guiboard.addArrow(point, pp.get(j));
+      } else // not in consider set
+      {
+        assert (value.charAt(0) == 'x');
+        m_guiboard.setAlphaColor(point, Color.gray);
+      }
+    }
+  }
+
+  // ------------------------------------------------------------
+
+  // Remove keyboard focus from all text components, so that
+  // keyboard shortcuts can work.
+  private void unFocus() {
+    KeyboardFocusManager.getCurrentKeyboardFocusManager().clearGlobalFocusOwner();
+  }
+
+  /** Callback from GuiBoard. Handle a mouse click. */
+  public void panelClicked() {
+    unFocus();
+  }
+
+  /** Callback from GuiBoard. Handle a mouse click. */
+  public void fieldClicked(HexPoint point, boolean ctrl, boolean shift) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.